http/api_handler.go

```diff
@@ -262,6 +291,14 @@ func (h *APIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// this for loop is equivalent to the if statements below
```

Comment:
This is a really cool idea. We've been meaning to have a router here for a while. I think as we have it now, callers of `NewAPIHandler` could prevent any of the other routes from being handled. What do you think about adding each of the handlers on APIHandler to the `ResourceHandler`s and potentially panicking/erring if someone tries to use a reserved route?

Reply:
Yah, that would be my suggestion actually. I love the `chi` router. Super expressive and has zero 3rd party dependencies, and is plenty fast. It would be trivial to mount these. I think having duplicates is panicable, makes sense to me. Not sure where that would happen though 🤔. Adding same http server twice maybe?

Reply:
@desa I'd be happy to break the handlers out into their own beasts. I'd like to do it after this body of work though. Hate to grow the scope of this atm. Would it be good to track this with a separate issue if we do it later?

Reply:
Would be really cool. Does chi support routing to prefixes? For example, if I have

`/api/v2/orgs/:id/owners` can I register `/api/v2/orgs` with chi and have it route to it?

Reply:
sure does 👌, those are table stakes in `chi` world lol

Reply:
@desa I put together some of the bits to make that happen here, cleans it up a fair amount. Check out the origins here:

https://github.com/influxdata/influxdb/pull/15758/files#diff-f047f7de48d9a985ade16057d9d48739R126

Then its use here: https://github.com/influxdata/influxdb/pull/15758/files#diff-f047f7de48d9a985ade16057d9d48739R410

Reply:
hell yeah!